### ISSUE TITLE
Graphql Version Confict in 4.16.x

### ISSUE THREAD

**Issue by DerMambo** [strapi/strapi#19114]
### Required System information

- Node.js version: v18.18.2
- NPM version:9.8.1
- Strapi version:4.16.2
- Database:sqlite
- Operating system:mac
- Is your project Javascript or Typescript:TS

### Describe the bug

After upgrading from 4.15 I got an error message after starting strapi

```
 Error: Cannot use GraphQLScalarType "Time" from another module or realm.                                      │
│                                                                                                                 │
│   Ensure that there is only one instance of "graphql" in the node_modules                                       │
│   directory. If different versions of "graphql" are the dependencies of other                                   │
│   relied on modules, use "resolutions" to ensure only one version is installed. 
```

### Steps to reproduce the behavior

1. Update to 4.16.2 having plugin-graphql installed

### Expected behavior

Strapi should start up as normal

### Additional context

My assumption is that 4.16.2 uses
```
    "graphql": "^15.5.1",
    "graphql-upload": "15.0.2",
```

Though graphql-upload 15.0.2 has a peer dependency on graphql 16.3.0.

Therefor graphql is installed two times.

**Comment by derrickmehaffy** [comment]
Related PR: https://github.com/strapi/strapi/pull/18811

**Comment by iwarner** [comment]
I am getting this error still - latest 4.16.2

Issue seems to be that strapi/plugin-graphql@4.16.2 - requires GraphQL graphql@15.8.0
Which in turns requires "@graphql-tools/schema": "8.5.1",
Which needs graphql@16 ...

See
https://github.com/strapi/strapi/blob/v4.16.2/packages/plugins/graphql/package.json

**Comment by 0xboomie** [comment]
I am also having the same issue

**Comment by iwarner** [comment]
I added in the package for Strapi

  "overrides": {
    "graphql": "^16"
  },
  
  and this seemed to work - a lot of the answers gave resolutions which are yarn - npm now offer overides :)

**Comment by fcsonline** [comment]
I am also having the same issue. Moving back to `4.15.5`.

**Comment by samuelperezcode** [comment]
Same issue

**Comment by benwmx** [comment]
I have the same problem in : 

- Node.js version: v20.10.0
- NPM version: v10.2.3 
- Strapi version:4.16.2

**Comment by swu223** [comment]
Also have the same problem. 

- Node: v18.19.0
- NPM: 10.2.5
- strapi: 4.16.2

**Comment by derrickmehaffy** [comment]
We know the issue exists, there is a workaround in the response above.

Locking this for now until engineering can work on it.

**Comment by dennism501** [comment]
Internal TID: 5693

**Comment by Bassel17** [comment]
This issue was fixed with this PR: #19182
